### ISSUE TITLE
[Bug] GGUF Q6_K export writes NaN block scales for all-zero blocks (0/0 in make_qx_quants_chunk) — NaN logits in llama.cpp

### ISSUE THREAD

**Issue by Entrpi** [intel/auto-round#1908]
## Summary

The GGUF Q6_K export path writes **NaN block scales** (the fp16 `d` at the end of each 210-byte Q6_K block) for blocks whose source weights are **all zero** — e.g. padded/unused vocab rows in an embedding tensor. A single NaN scale poisons the whole forward pass: the GGUF loads fine, but llama.cpp inference returns NaN logits/PPL for any batch touching those rows.

## Environment

- auto-round **v0.13.0** (also present on current `main` — the affected code is unchanged)
- Observed with the CLI:

```bash
auto-round --model google/gemma-4-26B-A4B-it-qat-q4_0-unquantized \
  --format "gguf:q4_0" --iters 200 --nsamples 256 --seqlen 2048
```

## Observed behavior

The exported model's `token_embd.weight` (Q6_K, 2,883,584 blocks) contained **exactly 17 NaN `d` scales**. Every one was on a block whose quantized payload was uniformly zero (verified: payload min=max=0). Affected token rows included 107, 528, 532, 558, and a cluster at 236770–236832 — i.e. unused/padded vocab entries whose embedding rows are all zeros.

Patching the 17 scales to `0.0` in the GGUF fully repairs the model — it then scores normal perplexity. So the expected behavior is clear: **all-zero blocks should get `d = 0.0`** (exact reconstruction).

## Root cause

`make_qx_quants_chunk` in `auto_round/export/export_to_gguf/packing.py` (used for Q6_K scale search via `search_gguf_scale_min_sym`) divides without a zero guard:

https://github.com/intel/auto-round/blob/v0.13.0/auto_round/export/export_to_gguf/packing.py#L195-L197

```python
sumlx = (w * chunk * L).sum(dim=-1)
suml2 = (w * L * L).sum(dim=-1)
scales = sumlx / suml2          # 0/0 = NaN for an all-zero group (w = chunk**2 = 0, L = 0)
```

For an all-zero 16-element group, `w = chunk*chunk = 0` and `L = 0`, so this is `0/0 = NaN`. The NaN then slips through every downstream guard:

1. `quant_tensor_gguf_sym_dq` (`auto_round/data_type/gguf.py:818`): `torch.where(torch.abs(scale) < 1e-30, 0, scale)` — `abs(NaN) < 1e-30` is `False`, so the NaN passes through.
2. `double_quant_tensor_sym_rtn` (`auto_round/data_type/gguf.py:242-246`): `argmax` over the NaN-containing abs picks the NaN → `wmax = NaN` → `d_scale = NaN`.
3. `q6_k_quant_block` writes `d_scale` as the block's fp16 `d`.

The refinement loop inside `make_qx_quants_chunk` never repairs it either, because `replace_id = (suml2_tmp > 0) & ...` is `False` when `suml2_tmp` is NaN.

Notably, the original non-chunked `make_qx_quants` (same file, line 270) already has the correct guard — `scales = sumlx * get_reciprocal(suml2)`, where `get_reciprocal` returns 0 for (near-)zero inputs. The chunked rewrite dropped it. llama.cpp's own `make_qx_quants` likewise guards this case (`if (!amax) { scale = 0; ... }`).

## Minimal reproduction (no model needed)

```python
import torch
from auto_round.data_type.gguf import quant_tensor_gguf_sym_dq

t = torch.zeros(2, 256)        # block 0: all zeros (e.g. an unused vocab row)
t[1] = torch.randn(256)
qdq, scales, _ = quant_tensor_gguf_sym_dq(t, bits=6, scale_dtype=torch.float32)
print(scales["scale"][0].flatten()[:4])   # tensor([nan, nan, nan, nan])
print(scales["d_scale"].flatten())        # tensor([nan, ...])
print(torch.isnan(qdq[0]).any())          # True — qdq weights are NaN too
```

## Fix (verified)

One line in `make_qx_quants_chunk`, matching the existing non-chunked implementation:

```python
scales = sumlx * get_reciprocal(suml2)
```

With this change the reproduction above yields sub-scales `0.0`, `d_scale = 0.0`, and exact all-zero reconstruction. Happy to send a PR.

Two adjacent spots probably deserve the same audit:
- `make_qp_quants` (`packing.py:491`) ends with an unguarded `return sumlx / suml2, L` (imatrix super-block path).
- The NaN-transparent `torch.where(torch.abs(scale) < 1e-30, ...)` in `quant_tensor_gguf_sym_dq` could be hardened with `torch.nan_to_num` as defense in depth.
